### ISSUE TITLE
Rule Request: Verify that #imageLiteral references existing asset

### ISSUE THREAD

**Issue by LinusU** [realm/SwiftLint#2512]
### New rule request

> 1\) Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

Referencing a non-existing asset with `#imageLiteral()` will result in a crash at runtime. As seen e.g. [here](https://github.com/realm/SwiftLint/issues/1587#issuecomment-304867157) and as experienced by me today, it's very surprising that `#imageLiteral()` is not checked by the Swift compiler in the first place. I think the next best thing after that would be to have an error lint about it, which is why I created this suggestion ☺️ 

> 2\) Provide several examples of what _would_ and _wouldn't_ trigger violations.

The following code would trigger a violation:

```swift
#imageLiteral(resourceNamed: "DoesNotExists")
```

The following code would not trigger a violation:
```swift
#imageLiteral(resourceNamed: "DoesExists")
```

> 3\) Should the rule be configurable, if so what parameters should be configurable?

No.

> 4\) Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

I think the rule should be enabled by default (unless the implementation turns out to be slow) since it's a problem that will cause a crash at runtime, and I don't see that there will be any false positives.

-----

I'd be willing to help contribute code for this ☺️

**Comment by skagedal** [comment]
This is indeed a surprising and annoying feature of image literals! To correctly check for this, you would have to check that the asset catalog containing the image is available in the target(s) that use the image literal. But maybe some simplified heuristics would be enough for many users. 

I would still recommend using [SwiftGen](https://github.com/SwiftGen/SwiftGen) instead for a more stable solution to this problem.

**Comment by LinusU** [comment]
> I would still recommend using SwiftGen instead for a more stable solution to this problem.

Personally, I prefer literals since they are statically analyzable, and doesn't require a separate tool for code generation. Each to their own though!

> But maybe some simplified heuristics would be enough for many users.

Tried some very simple heuristics in my project and it works really well, but we only have a single asset catalog though...
